### ISSUE TITLE
Git file with space raise exception

### ISSUE THREAD

**Issue by PTank** [dhruvkb/pls#39]
## Description

File in git repo with space, shlex.split() make a longer list than accepted by PLS.

## Reproduction

Create the file

```bash
mkdir test
cd test
git init
touch "file.txt badly formatted with more word than the 3 accepted by the code"
```

Just run  `pls` and it raise `ExecException: Could not parse Git status code.`

The status can be wrong if  `components == 3`

## Expectation

Do not raise exception, perhaps log warning. And have a better split.

## Resolution

- [x] 🙋 I would be interested in resolving this bug.

**Comment by PTank** [comment]
Ok, i'm on linux/ Ubuntu
git: 2.17.1

The output do not quote the filename.

```bash
touch "test.lol lol"
git status  --porcelain
?? test.lol lol
```

**Comment by PTank** [comment]
Work well with git: 2.37.1.

But sadly cannot use pls at work, can't add git ppa here.

Thanks

**Comment by dhruvkb** [comment]
Since Git v2.17.1 is quite old (2018), I won't be able to add support for it. But since it hasn't reached EOL, if you want to submit a pull request, I'll be happy to review and merge it.

**Comment by dhruvkb** [comment]
Thank you @PTank for finding and fixing the issue! It's been released in v5.4.0.
